### PR TITLE
Implement feature/SPF2-18: Warehouse Edit API

### PR DESCRIPTION
**Pull request by davidmaxxxy**
Implemented the backend API for editing warehouse details.

Changes made:

- Created the `updateWarehouseDetails` controller in `warehouseController.js` to handle PUT requests.

- Added data validation using Joi to ensure non-empty values and proper format.

- Integrated the new routes in `warehouseRoutes.js`.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
+app.get("/", (req, res) => {
+  res.send("Hello World!");
 });
 
-
-// Routes 
-app.use('/inventories/:warehouse_id', inventoriesRoutes);
-
+// Routes
+app.use("/inventories/", inventoriesRoutes);
```
Comment (bernard493) [inline server.js]: why did you change this line of code from app.use('/inventories/:warehouse_id', inventoriesRoutes);  to  app.use("/inventories/", inventoriesRoutes);

Comment (davidmaxxxy) [reply server.js]: Unintentionally. Probably changed it whilst dealing with warehouses routes.

Comment (bernard493) [reply server.js]: oh ok then you need to change it back cox to add an inventory you need to be able to know what warehouse that inventory belongs to that why i added the warehouse_id to the route

File: server.js [hunk 2]
```diff
@@ -1,27 +1,23 @@
...
+app.use("/warehouses/", warehouseRoutes);
```
Comment (bernard493) [inline server.js:19]: In your warehouseController, you get the warehouse_id from the req.params.id but in this server js file your route is not dynamically allowing a params in the route, which means req.params.id will be undefined  
and if you check your ticket on Jira you gonna see the expected route 

Fix 
```
server.js
app.use("/warehouses/:id", warehouseRoutes);

warehouseController.js
const warehouseId = parseInt(req.params.id);
```


If you modify your route to app.use("/warehouses/:id", warehouseRoutes);, then the API endpoint will only be accessible when the request includes a valid warehouse ID, such as warehouses/554421.

File: controllers/warehouseController.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+const knex = require("knex")(require("../knexfile"));
+
+const updateWarehouseDetails = async (req, res) => {
+  try {
+    const warehouseId = parseInt(req.params.id);
+
+    if (!warehouseId || isNaN(warehouseId)) {
+      return res.status(400).json({ message: "Invalid warehouse ID." });
+    }
+
```
Comment (bernard493) [inline controllers/warehouseController.js:11]: If you modify your route to app.use("/warehouses/:id", warehouseRoutes);, then the API endpoint will only be accessible when the request includes a valid warehouse ID, such as warehouses/554421. This eliminates the need to manually check if the ID was passed or if it's invalid in the controller.

File: controllers/warehouseController.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    if (!warehouseExists) {
+      return res.status(404).json({ message: "Warehouse not found." });
+    }
+
+    await knex("warehouses").where("id", warehouseId).update(req.body);
+
+    const updatedWarehouse = await knex("warehouses")
+      .where("id", warehouseId)
+      .first();
+
```
Comment (bernard493) [inline controllers/warehouseController.js:52]: You are retrieving the updated warehouse after executing the update query, but there is a chance the update operation could fail, and you are not checking whether it was successful. I recommend adding a check to ensure the update was successful before querying the database for the updated information. If the update fails, you can return an appropriate error response instead of making an additional DB call.

```

const updateCount   = await knex("warehouses").where("id", warehouseId).update(req.body);

// 0 ==  failed
//1
if (updateCount === 0) {
  return res.status(500).json({ message: "Update failed: Warehouse not found or no changes made. Try again later " });
}
```

**NOTE** 
check updateCount  to make sure its what am thinking
